lib/internal/modules/esm/module_job.js

```diff
@@ -109,13 +110,14 @@ class ModuleJob {
         if (format === 'commonjs') {
           const importStatement = splitStack[1];
           const namedImports = StringPrototypeMatch(importStatement, /{.*}/)[0];
+          const importsAsPattern = StringPrototypeReplace(namedImports, /\s+as\s+/g, ': ');
           e.message = `The requested module '${childSpecifier}' is expected ` +
             'to be of type CommonJS, which does not support named exports. ' +
             'CommonJS modules can be imported by importing the default ' +
             'export.\n' +
             'For example:\n' +
             `import pkg from '${childSpecifier}';\n` +
-            `const ${namedImports} = pkg;`;
+            `const ${importsAsPattern} = pkg;`;
```

Comment:
I don't find the name `importsAsPattern` very representative of what contains the variable. What about:
```suggestion
            `const ${destructuringAssignment} = pkg;`;
```

Reply:
Yes that sounds much better

Reply:
I've amended the commit to incorporate this rename, :+1: